pandas/io/packers.py

```diff
@@ -371,9 +385,8 @@ def encode(obj):
             return {'typ': 'block_manager',
                     'klass': obj.__class__.__name__,
                     'axes': data.axes,
-                    'blocks': [{'items': data.items.take(b.mgr_locs),
-                                'locs': b.mgr_locs.as_array,
-                                'values': convert(b.values),
+                    'blocks': [{'locs': b.mgr_locs.as_array,
+                                'values': convert_if_not_dt_index(b.values),
```

Comment:
actually I think you could trivially change ``core/internals.py/make_block`` to take a ``DatetimeTZDtype`` and correctly create it. Then you would have to make to serialize that properly (as a string is prob ok) I think (you would need to fix the reverse lookup as it only deals with numpy atm). 

I think the same problem exists with categorical, prob not testing it.